extension/modules/personas.js

```diff
@@ -0,0 +1,1480 @@
+        let theme = PersonaService.getPersonaJSON(persona);
+
+        let headerURI;
+        if (persona.custom) {
+            headerURI = theme.headerURL || theme.header;
+        } else {
+            headerURI = theme.dataurl || theme.iconURL;
+        }
+
+        item.setAttribute("class", "menuitem-iconic");
+        item.setAttribute("image", headerURI);
+        item.setAttribute("label", persona.name);
+        //    item.setAttribute("type", "checkbox");
+        item.setAttribute("checked", (PersonaService.selected != "default" &&
+            PersonaService.currentPersona &&
+            PersonaService.currentPersona.id == persona.id));
+        item.setAttribute("autocheck", "false");
+        item.addEventListener("command",function(event){PersonaController.onSelectPersona(event)},true);
+        item.setAttribute("recent", persona.recent ? "true" : "false");
+        item.setAttribute("persona", JSON.stringify(theme));
+        item.addEventListener("DOMMenuItemActive", function(evt) {
+            PersonaController.onPreviewPersona(evt)
+        }, false);
+        item.addEventListener("DOMMenuItemInactive", function(evt) {
+            PersonaController.onResetPersona(evt)
+        }, false);
+
+        return item;
+    },
+
+    _createViewMoreItem: function(category, categoryId, document) {
+
+        let item = document.createElement("menuitem");
+
+        item.setAttribute("class", "menuitem-iconic");
+        item.setAttribute("label", this._strings.get("viewMoreFrom", [category]));
+        item.addEventListener("command",function(event){PersonaController.openURLInTab(this.getAttribute('href'),event)},true);
+
+        if (categoryId == "featured") {
+            item.setAttribute("href", PersonaService.getURL("browse", {
+                SORT: "featured"
+            }));
+        }
+
+        return item;
+    },
+
+    _createRandomItem: function(aCategoryName, aCategory, document) {
+
+        let item = document.createElement("menuitem");
+
+        item.setAttribute("class", "menuitem-iconic");
+        item.setAttribute("label", this._strings.get("useRandomPersona", [aCategoryName]));
+        item.addEventListener("command",function(event){PersonaController.onSelectPersona(event)},true);
+        item.setAttribute("persona", "random");
+        item.setAttribute("category", aCategory || aCategoryName);
+
+        return item;
+    },
+
+    toggleFavoritesRotation: function() {
+        if (PersonaService.selected != "randomFavorite") {
+            PersonaService.changeToRandomFavoritePersona();
+        } else {
+            PersonaService.selected = "current";
+        }
+    },
+
+	THUNDERBIRD_ID: "{3550f703-e582-4d05-9a08-453d09bdfdc6}",
+	FIREFOX_ID: "{ec8030f7-c20a-464f-9b0e-13a3a9e97384}",
+	SEAMONKEY_ID:"{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}",	
+	appInfo: Components.classes["@mozilla.org/xre/app-info;1"].getService(Components.interfaces.nsIXULAppInfo),		
```

Comment:
Those constants are defined in `extension/modules/service.jsextension/modules/service.js`.